crates/solver/src/solver/paraswap_solver.rs

```diff
@@ -93,8 +93,12 @@ impl SingleOrderSolving for ParaswapSolver {
             .get_token_infos(&[order.sell_token, order.buy_token])
             .await;
         let (price_response, amount) = self.get_price_for_order(&order, &token_info).await?;
-        if !satisfies_limit_price(&order, &price_response) {
-            tracing::debug!("Order limit price not respected");
+        if !execution_respects_order(
+            &order,
+            price_response.src_amount,
+            price_response.dest_amount,
```

Comment:
TODO: Have to double check this is right. I think we might have to select based on order_kind here but in that case wasn't the original code wrong too?

Reply:
I think this is correct. `src_token` always refers to the token being sold and `dest_token` to the token being bought. Reference: https://developers.paraswap.network/api/get-rate-for-a-token-pair

Specifically the documentatino for the `amount` parameter makes it quite clear:
> srcToken amount (in case of SELL) or destToken amount (in case of BUY).